### PR TITLE
add buildah backend

### PR DESCRIPTION
**Pull request by TomasTomecek**
TODO:
* [x] all tests are passing

Next PR:
* [ ] create interface for commit() (needed by bender)

### CODE REVIEW

**Comment by TomasTomecek** [comment]
I just removed conu from codacy.

**Comment by TomasTomecek** [comment]
fingers crossed for :green_apple: CI

**Comment by TomasTomecek** [comment]
all tests are finally green for me locally

**Comment by TomasTomecek** [comment]
CI passed \o/

**Comment by rpitonak** [comment]
I will try to do a review this week.

File: conu/backend/buildah/backend.py [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+        raw_version = run_cmd(["buildah", "version"], return_output=True)
+        regex = re.compile(r"Version:\s*(\d+)\.(\d+)\.(\d+)")
+        match = regex.findall(raw_version)
+        try:
+            return match[0]
+        except IndexError:
+            logger.error("unable to parse version from `buildah version`")
+            return
+
+    def list_containers(self):
```
[the remark this answers is not in the document]
Comment (TomasTomecek) [reply conu/backend/buildah/backend.py]: nice! I did not realize that buildah would not return an empty list:
```
$ buildah ps -a --json
null

$ ipython3
Python 3.7.4 (default, Jul  9 2019, 16:32:37)
Type 'copyright', 'credits' or 'license' for more information
IPython 7.2.0 -- An enhanced Interactive Python. Type '?' for help.
Exception reporting mode: Verbose

In [1]: import json

In [2]: json.loads("null")

In [3]: print(json.loads("null"))
None
```

pushed a fix for this

**Comment by TomasTomecek** [comment]
feel free to merge once the tests pass

**Pull request merged**